app/src/main/kotlin/cz/covid19cz/app/service/CovidService.kt

```diff
@@ -73,185 +70,104 @@ class CovidService : Service() {
             // null intent is in case service is restarted by system
             ACTION_START, null -> {
                 createNotification()
-
-                startBleAdvertising()
-                startBleScanning()
-                wakeLock(true)
+                subscribeToBluetoothAndLocationStates()
+                wakeLockManager.acquire()
             }
             ACTION_STOP -> {
-                wakeLock(false)
+                wakeLockManager.release()
                 stopForeground(true)
                 stopSelf()
             }
             ACTION_UPDATE -> {
                 createNotification()
+                if (isLocationEnabled() && btUtils.isBtEnabled()) {
+                    turnMaskOn()
+                } else {
+                   turnMaskOff()
+                }
             }
         }
         return START_STICKY
     }
 
-    fun createNotification() {
-
-        val btEnabled = btUtils.isBtEnabled()
-        val locationEnabled = isLocationEnabled()
-
-        createNotificationChannel()
-        val notificationIntent = Intent(this, MainActivity::class.java)
-        val pendingIntent = PendingIntent.getActivity(
-            this,
-            0, notificationIntent, 0
-        )
+    override fun onDestroy() {
+        wakeLockManager.release()
+        turnMaskOff()
+        unsubscribeFromBluetoothAndLocationStates()
 
-        val title :Int
-        val text : Int
-        val icon :Int
+        super.onDestroy()
+    }
 
-        //TODO: work in progress
-        //if (btEnabled && locationEnabled){
-        if (true){
-            title = R.string.notification_title
-            text = R.string.notification_text
-            icon = R.drawable.ic_notification_normal
-        } else {
-            title = R.string.notification_title_error
-            text = R.string.notification_text_error
-            icon = R.drawable.ic_notification_error
-        }
+    override fun onBind(intent: Intent?): IBinder? {
+        return null
+    }
 
-        val notification = NotificationCompat.Builder(this, CHANNEL_ID)
-            .setContentTitle(getString(title))
-            .setContentText(getString(text))
-            .setSmallIcon(icon)
-            .setContentIntent(pendingIntent)
-            .build();
-        startForeground(1, notification)
+    private fun turnMaskOn() {
+        startBleAdvertising()
+        startBleScanning()
     }
 
-    override fun onDestroy() {
+    private fun turnMaskOff() {
         btUtils.stopScanning()
         btUtils.stopAdvertising()
+
```

Comment:
Nechtelo by to tady ulozit to co je zatim namereno do databaze? Nebo jsou nekompletni mereni bezcenny?